src/mca/gds/base/gds_base_fns.c

```diff
@@ -210,24 +259,120 @@ pmix_status_t pmix_gds_base_store_modex(struct pmix_namespace_t *nspace,
                     &bkt, &bo2, &cnt, PMIX_BYTE_OBJECT);
         }
         PMIX_DESTRUCT(&bkt);
+
         if (PMIX_ERR_UNPACK_READ_PAST_END_OF_BUFFER == rc) {
             rc = PMIX_SUCCESS;
         } else if (PMIX_SUCCESS != rc) {
-            goto error;
+            PMIX_ERROR_LOG(rc);
+            goto exit;
         }
         /* unpack and process the next blob */
         cnt = 1;
         PMIX_BFROPS_UNPACK(rc, pmix_globals.mypeer,
                 buff, &bo, &cnt, PMIX_BYTE_OBJECT);
     }
+
     if (PMIX_ERR_UNPACK_READ_PAST_END_OF_BUFFER == rc) {
         rc = PMIX_SUCCESS;
+    } else if (PMIX_SUCCESS != rc) {
+        PMIX_ERROR_LOG(rc);
+    }
+exit:
+    if (pmix_argv_count(kmap) > 0) {
```

Comment:
You don't need the extra call here - if kmap != NULL will suffice and save a little time

Reply:
Fixed